Rakefile

```diff
@@ -4,16 +4,38 @@
 
 protoc_binary="protoc"
 protoc_version="3.5.1"
-gogo_dir="/tmp/gogo"
+gogo_dir=Dir.pwd + "/toolchains/gogo"
+gogo_bin=Dir.pwd + "/toolchains/gogo-bin-v1.3.2"
 protoc_binary_2="protoc"
 protoc_version_2="21.12"
 protoc_gen_go_dir="/tmp/protoc-gen-go"
 protoc_jsonschema_version="73d5723"
 
 
 namespace :codegen do
+  task :clean do
+      sh "rm -rf #{gogo_dir}"
+  end
+
+  task :setup_gogo => ['install_protoc'] do
+    if ! Dir.exist?(gogo_dir) then
```

Comment:
We might want to try to stay in Ruby land probably even if the old code didn't do that. Eg:
```ruby
Dir.mkdir(gogo_dir) unless  File.directory?(gogo_dir) #Maybe FileUtils.mkdir_p though
# <other stuff >
```